activerecord/lib/active_record/counter_cache.rb

```diff
@@ -73,13 +88,29 @@ def reset_counters(id, *counters)
       #   # UPDATE posts
       #   #    SET comment_count = COALESCE(comment_count, 0) + 1
       #   #  WHERE id IN (10, 15)
+      #
+      #   # For the Posts with id of 10 and 15, increment the comment_count by 1
+      #   # and update the updated_at value for each counter.
+      #   Post.update_counters [10, 15], comment_count: 1, touch: true
+      #   # Executes the following SQL:
+      #   # UPDATE posts
+      #   #    SET comment_count = COALESCE(comment_count, 0) + 1,
+      #   #    `updated_at` = '2016-10-13T09:59:23-05:00'
+      #   #  WHERE id IN (10, 15)
       def update_counters(id, counters)
+        touch_time = Time.now
+        timestamps = resolve_timestamp_columns(find(id), counters.delete(:touch))
```

Comment:
This method is now doing a query even when `touch` is not passed. I think we should only do that if touch is provided.